library/src/main/java/com/datatorrent/lib/state/BucketedState.java

```diff
@@ -29,11 +31,17 @@
 public interface BucketedState
 {
   /**
+   * An expired value. In some implementations where bucketId is time related then the event can be old and
+   * the get methods- getSync & getAsync return this special slice instance.
+   */
+  Slice EXPIRED = new Slice(Ints.toByteArray(-1));
```

Comment:
I can use Slice(null, -1, -1). In the usages, comparisons should be made using ```==``` and not ```equals```.

For every expired event I didn't want to instantiate a Slice object. So even if I extend Slice, I will re-use a special instance of slice so thought why not use just a special instance.

There is a problem with all the approaches I can think of which is that Slice cannot have an immutable extension because all the members are publicly exposed without getters and setters.